api/src/main/java/org/apache/iceberg/transforms/Identity.java

```diff
@@ -39,7 +42,7 @@ class Identity<T> implements Transform<T, T> {
   @Deprecated
   public static <I> Identity<I> get(Type type) {
     Preconditions.checkArgument(
-        type.typeId() != Type.TypeID.VARIANT, "Unsupported type for identity: %s", type);
+        !UNSUPPORTED_TYPES.contains(type), "Unsupported type for identity: %s", type);
```

Comment:
Why change this? It works just fine right? We can alter this when we add another type.

Reply:
Yes. It works fine. I thought you mentioned that you wanted to make such change in a previous PR. :)